tests/python/unittest/test_sparse_operator.py

```diff
@@ -135,11 +133,12 @@ def test_dot_csr_dns_rsp(csr_shape, dns_shape, dns_grad_stype, trans_csr):
         location = {'lhs':csr, 'rhs':dns2, 'zero':mx.nd.zeros(rsp_expected.shape)}
         expected = {'rhs':rhs_backward_grad, 'zero':out_np}
         # dot(lhs, rhs) + zeros
-        check_symbolic_forward(test, location, [rsp_expected.asnumpy()])
+        check_symbolic_forward(test, location, [rsp_expected.asnumpy()], rtol=1e-3, atol=1e-4)
         check_symbolic_backward(test, location, [out_np], expected,
-                                grad_req={'lhs': 'null', 'rhs': 'write', 'zero' : 'write'})
+                                grad_req={'lhs': 'null', 'rhs': 'write', 'zero' : 'write'},
+                                rtol=1e-3, atol=1e-4)
 
-    lhs_shape = (rnd.randint(1, 10),rnd.randint(1, 10))
+    lhs_shape = (rnd.randint(1, 10), rnd.randint(1, 10))
     test_dot_csr_dns_rsp(lhs_shape, (lhs_shape[1], rnd.randint(1, 10)), 'row_sparse', False)
     test_dot_csr_dns_rsp(lhs_shape, (lhs_shape[0], rnd.randint(1, 10)), 'row_sparse', True)
     test_dot_csr_dns_rsp(lhs_shape, (lhs_shape[1], rnd.randint(1, 10)), 'default', False)
```

Comment:
Could you also add test dot csr_dns_dns, and change infer storage for dot to be: 
```
csr.T, dns => rsp
csr, dns => dns 
```
?